### ISSUE TITLE
-Wdocumentation contains too many unrelated elements under one umbrella

### ISSUE THREAD

**Issue by JVApen** [llvm/llvm-project#71155]
I'm looking at enabling the `-Wdocumentation` warning on a large project and as expected this gives many violations.
To enable these warnings, it would be beneficial if a separate warning would control every category of warnings.

For example:
````
/// @param[in] i First param
/// @param[in] j
auto f(int i, int j) noexcept -> bool;
````
This outputs `2:16: warning: empty paragraph passed to '@param' command [-Wdocumentation]`

However, in our codebase, this warning gets triggered a lot since doxygen gives warnings if the param `j` is not mentioned while another parameter is documented. So to silence those warnings, these empty paragraphs are added.

Many other messages under this umbrella can be found on https://clang.llvm.org/docs/DiagnosticsReference.html#wdocumentation
